### PR TITLE
Add Sentence/Title cases (WIP) (TESTING)

### LINKED ISSUE #13: Change Case
Currently PecoRename only allows UPPERCASE/lowercase . 
Few more case like : 
Sentence case
Title Case
tOGGLE cASE could be added .

### PR DESCRIPTION
**Pull request by bach5000**
Should fix #13.

### CODE REVIEW

File: Main/Renamer_UpperLower.cpp [hunk 1]
```diff
@@ -61,18 +62,40 @@ void Renamer_UpperLower::RenameList(BList *FileList) {
...
-				} else { // kleinbuchstaben
-					if (((Buchstabe>=0x41) && (Buchstabe<=0x5a)) || ( (Byte0==0xc3) && (Buchstabe>=0x80 && Buchstabe<=0x96) || (Buchstabe>=0x98 && Buchstabe<=0x9e)))
+						break;
+					case 1: if (((Buchstabe>=0x41) && (Buchstabe<=0x5a)) || ( (Byte0==0xc3) && (Buchstabe>=0x80 && Buchstabe<=0x96) || (Buchstabe>=0x98 && Buchstabe<=0x9e)))
 						Buchstabe |= 32;
+						break;
+					case 2: 
+						// Capitalize only the first letter.
+						if (isCap == 1)
+							{isCap = 0; Buchstabe &= ~32;}
```
Comment (janus2) [inline Main/Renamer_UpperLower.cpp]: You should check the range of the character (Bunchstabe) if it is not an appropriate letter you may transform a symbol in something else.

File: Main/Renamer_UpperLower.cpp [hunk 2]
```diff
@@ -61,18 +62,40 @@ void Renamer_UpperLower::RenameList(BList *FileList) {
...
+						else {
+							if (((Buchstabe>=0x41) && (Buchstabe<=0x5a)) || ( (Byte0==0xc3) && (Buchstabe>=0x80 && Buchstabe<=0x96) || (Buchstabe>=0x98 && Buchstabe<=0x9e)))
+								Buchstabe |= 32;
+						}
+						break;
+					case 3:
+						if (isCap == 1) {isCap = 0; Buchstabe &= ~32;}
```
Comment (janus2) [inline Main/Renamer_UpperLower.cpp]: You should check the range of the character (Bunchstabe) if it is not an appropriate letter you may transform a symbol in something else.

File: Main/Renamer_UpperLower.cpp [hunk 3]
```diff
@@ -61,18 +62,40 @@ void Renamer_UpperLower::RenameList(BList *FileList) {
 		char	*tempStr = new char[laenge + 1];
 
 		uint8	Buchstabe, Byte0 = 0;
-		bool	MakeUpper = !bool(fUpperOrLower->Menu()->IndexOf(fUpperOrLower->Menu()->FindMarked()));
+		int32   choice = fUpperOrLower->Menu()->IndexOf(fUpperOrLower->Menu()->FindMarked());
+	
+		int32 isCap = 1;
```
Comment (janus2) [inline Main/Renamer_UpperLower.cpp]: Can this be a boolean? For example bool toCapitalize

File: Main/Renamer_UpperLower.cpp [hunk 4]
```diff
@@ -61,18 +62,40 @@ void Renamer_UpperLower::RenameList(BList *FileList) {
...
+		
 		for (int j = 0; j < laenge; j++) {
 			Buchstabe = ListItem->fName.ByteAt(j);
 			if (Buchstabe>=0xc0) {
 				Byte0 = Buchstabe;
 			} else {
-				if (MakeUpper) {
-					if (((Buchstabe>=0x61) && (Buchstabe<=0x7a)) || ( (Byte0==0xc3) && (Buchstabe>=0xa0 && Buchstabe<=0xb6) || (Buchstabe>=0xb8 && Buchstabe<=0xbe)))
+				switch (choice){
+					case 0: if (((Buchstabe>=0x61) && (Buchstabe<=0x7a)) || ( (Byte0==0xc3) && (Buchstabe>=0xa0 && Buchstabe<=0xb6) || (Buchstabe>=0xb8 && Buchstabe<=0xbe)))
```
Comment (janus2) [inline Main/Renamer_UpperLower.cpp]: can this be a function for example capitalizeLetter() that take Byte0 and Buchstabe as parameters and return the new Bunchstabe?

**Comment by janus2** [comment]
Just verify that our range check  is not very good a lot of symbols change during manipulation.
In the future we must found a better solution, but it must be fast because these manipulations occurred on each keystroke possibly on thousands of names.

File: Main/Renamer_UpperLower.cpp (hunk 3 above)
Comment (bach5000) [reply Main/Renamer_UpperLower.cpp]: @janus Okay.

File: Main/Renamer_UpperLower.cpp (hunk 4 above)
Comment (bach5000) [reply Main/Renamer_UpperLower.cpp]: @janus2 so do you want me to put the entire `switch` statement into a function? Or do you want me to use BeOS's functions?

**Comment by scottmc** [comment]
What's the status on this one?

**Comment by humdingerb** [comment]
I think this PR is obsolete. PR #40 fixed issue #13.

**Comment by janus2** [comment]
Yes we can close this.

**Pull request closed without merging**